runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -137,8 +137,8 @@ def __init__(
             **kwargs.get("launched_properties", {}),
         }
         self._docker_user = None
-        self._namespace = kwargs.get("namespace")
-        self._context = kwargs.get("context")
+        self._namespace = self.compute_properties.get("namespace")
+        self._context = self.compute_properties.get("context")
```

Comment:
I think it should be `kwargs.get("namespace") or self.compute_properties.get("namespace")`, to handle case of passing in namespace/context through cluster factory vs reloading from den